.azure/vs2022.yml

```diff
@@ -1540,9 +1545,121 @@ stages:
           - publish: $(Build.StagingDirectory)
             artifact: devtools-$(arch)
 
+  - stage: swift_format
+    dependsOn: [toolchain, sdk, devtools]
+    jobs:
+      - job: build
+        strategy:
+          matrix:
+            'x64':
+              arch: amd64
+              platform: x86_64
+              binDir: bin64
+            # TODO(compnerd) enable AArch64 toolchain
+            # 'arm64':
+            #   arch: arm64
+            #   platform: aarch64
+            #   binDir: bin64a
+        steps:
+          - download: current
+            artifact: toolchain-amd64
+          - download: current
+            artifact: windows-sdk-$(arch)
+          - download: current
+            artifact: devtools-amd64
+          - checkout: apple/swift-format
+            fetchDepth: 1
+          - task: PowerShell@2
+            inputs:
+              targetType: 'inline'
+              script: |
+                $LibraryRoot = "$(Agent.BuildDirectory)/Library"
+                $ToolchainInstallRoot = "$LibraryRoot/Developer/Toolchains/unknown-Asserts-development.xctoolchain"
+                $PlatformInstallRoot = "$LibraryRoot/Developer/Platforms/Windows.platform"
+                $SDKInstallRoot = "$PlatformInstallRoot/Developer/SDKs/Windows.sdk"
+                $ToolchainBinaryCache = "$(Agent.BuildDirectory)/toolchain-amd64"
+                $DevtoolsBinaryCache = "$(Agent.BuildDirectory)/devtools-amd64"
+                $PlatformBinaryCache = "$(Agent.BuildDirectory)/windows-sdk-$(arch)"
+
+                $SDKSourceRoot = "$PlatformBinaryCache/Library/Developer/Platforms/Windows.platform/Developer/SDKs/Windows.sdk"
+                $XCTestSourceRoot = "$PlatformBinaryCache/Library/Developer/Platforms/Windows.platform/Developer/Library/XCTest-development"
+
+                function Copy-File($Src, $Dst) {
+                  $DstDir = [IO.Path]::GetDirectoryName($Dst)
+                  New-Item -ItemType Directory -ErrorAction Ignore $DstDir | Out-Null
+                  Copy-Item -Force $Src $Dst
+                }
+
+                function Copy-Directory($Src, $Dst) {
+                  New-Item -ItemType Directory -ErrorAction Ignore $Dst | Out-Null
+                  Copy-Item -Force -Recurse $Src $Dst
```

Comment:
I think that we want to move instead - we are space constrained on the hosted runners.

Reply:
Ok. And it is also faster than copying. Speed for free!